storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1382,5 +1422,438 @@ public static TopologyInfo getTopologyInfo(String name, String asUser, Map storm
     public static int toPositive(int number) {
         return number & Integer.MAX_VALUE;
     }
-}
 
+
+
+    //Everything from here on is translated from the old util.clj (storm-core/src/clj/backtype.storm/util.clj)
+
+    //Wraps an exception in a RuntimeException if needed
+    public static Exception wrapInRuntime (Exception e) {
+        if (e instanceof RuntimeException) {
+            return e;
+        } else {
+            return (new RuntimeException(e));
+        }
+    }
+
+    public static final boolean isOnWindows = "Windows_NT".equals(System.getenv("OS"));
+
+    public static final String filePathSeparator = System.getProperty("file.separator");
+
+    public static final String classPathSeparator = System.getProperty("path.separator");
+
+
+    /*
+        Returns the first item of coll for which (pred item) returns logical true.
+        Consumes sequences up to the first match, will consume the entire sequence
+        and return nil if no match is found.
+     */
+    public static Object findFirst (IPredicate pred, Collection coll) {
+        if (coll == null || pred == null) {
+            return null;
+        } else {
+            Iterator<Object> iter = coll.iterator();
+            if (iter==null || !iter.hasNext()) {
+                return null;
+            } else {
+                do {
+                    Object obj = iter.next();
+                    if (pred.test(obj)) {
```

Comment:
The interface for pred returns a `Boolean` not a `boolean` as such an NPE is possible here.  Can we switch it to little b boolean to avoid this?